File: test/language/templates/cpp98/InstantiateUnusedTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_unused/U32.h"

namespace templates
{
namespace instantiate_unused
{

TEST(InstantiateUnusedTest, readWrite)
{
    U32 u32; // check that unused template is instantiated via the instantiate command
    u32.setValue(13);

    zserio::BitStreamWriter writer;
    u32.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    U32 readU32(reader);

    ASSERT_TRUE(u32 == readU32);
}

} // namespace instantiate_unused
} // namespace templates

File: test/language/templates/java/templates/InstantiateUnusedTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamWriter;

import templates.instantiate_unused.U32;

public class InstantiateUnusedTest
{
    @Test
    public void readWrite() throws IOException
    {
        final U32 u32 = new U32(13); // check that unused template is instantiated via the instantiate command

        final ByteArrayBitStreamWriter writer = new ByteArrayBitStreamWriter();
        u32.write(writer);

        final BitStreamReader reader = new ByteArrayBitStreamReader(
                writer.toByteArray(), writer.getBitPosition());
        final U32 readU32 = new U32(reader);
        assertTrue(u32.equals(readU32));
    }
}
